src/shared/featureToggle.ts

```diff
@@ -3,15 +3,20 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { SettingsConfiguration } from './settingsConfiguration'
+import { extensionSettingsPrefix } from './constants'
+import { SettingsConfiguration, DefaultSettingsConfiguration } from './settingsConfiguration'
 
-// This enum represents the list of active feature flags to check for
-// Features all need to be added with matching enums and string values
-// e.g. `Feature1 = 'Feature1'`
-// Features that do not follow this scheme will not work.
-// You cannot have more active features than FeatureToggle.maxFeatures (default: 5)
-// Any features that are flagged in the code but not added here will always return false.
-export enum ActiveFeatureKeys {}
+/**
+ * This enum represents the list of active feature flags to check for
+ * Features all need to be added with matching enums and string values
+ * e.g. `Feature1 = 'Feature1'`
+ * Features that do not follow this scheme will not work.
+ * You cannot have more active features than FeatureToggle.maxFeatures (default: 5)
+ * Any features that are flagged in the code but not added here will always return false.
+ */
+export enum ActiveFeatureKeys {
```

Comment:
Do we use enums elsewhere? I think we should stick to union types.

```typescript
export type ActiveFeatureKeys = 'CloudWatchLogs' | '...' | '...'
```

From what I can see, enums are generally avoided in typescript.

Reply:
> We do use enums in a few other places. I think I opted for enums because I liked the simplicity of specifying `ActiveFeatureKeys.CloudWatchLogs` etc. elsewhere.

Also we end up defining constants for the union values, which is a bit redundant...

Reply:
Yeah it's weird because the type unions give you basically the same value.

You get autocomplete and compile-time type checking. There's not really a reason to define constants for the items in the union. The strings *are* the constants. You just can't do `functionThatAcceptsEnum(MyEnum.<autocomplete>)` you just do `functionThatAcceptsUnion('<autocomplete>`) instead (i.e. start typing a string).

Either way works though.

Reply:
From an offline conversation: I don't think union types let you programmatically access the types (via something like `Object.keys`), so you can't build a map off of them